hexa/datasets/schema/queries.py

```diff
@@ -37,6 +42,21 @@ def resolve_dataset_version(_, info, **kwargs):
         return None
 
 
+@datasets_queries.field("datasetFileSnapshot")
+def resolve_dataset_file_snapshot(_, info, **kwargs):
+    try:
+        if kwargs.get("file_id"):
+            return DatasetFileMetadata.objects.get(
```

Comment:
I'm pretty sure that you have multiple `DatasetFileMetadata` per `DatasetVersionFile`

Reply:
I was in doubt for this one as it would make sense as well to update the content of the sample instead of creating  a new one.  Would one make use of a history of file samples?

Reply:
Let's start simple with a FK on `DatasetVersionFile` to `DatasetFileMetadata`